zend_abstract_interface/headers/php5/headers.c

```diff
@@ -0,0 +1,60 @@
+#include "../headers.h"
+
+#include <php.h>
+#include <zai_assert/zai_assert.h>
+
+zai_header_result zai_read_header(zai_string_view uppercase_header_name, zai_string_view *header_value TSRMLS_DC) {
```

Comment:
Philosophical question: should we set `*header_value` to the empty string on entry?